yarn-project/types/src/interfaces/low_nullifier_witness.ts

```diff
@@ -0,0 +1,41 @@
+import { Fr, NULLIFIER_TREE_HEIGHT } from '@aztec/circuits.js';
+
+import { SiblingPath } from '../sibling_path.js';
+import { LeafData } from './leaf_data.js';
+
+/**
+ * Low nullifier membership witness.
+ * @remarks Low nullifier membership witness can be used to perform a nullifier non-inclusion proof by leveraging
+ * the "linked list structure" of leaves and proving that a lower nullifier is pointing to a bigger next value than
+ * the nullifier we are trying to prove non-inclusion for.
+ */
+export class LowNullifierMembershipWitness {
```

Comment:
Could / Should the existing membership witness implementation be used, but make it generic over the underlying type? 

Just a thought, no requirement if its more messy

Reply:
They are currently quite different because MembershipWitness does not contain leafData which is needed in the nullifier's case because we also need to prove that leafData.value is actually the nullifier we are looking for so I would keep them separate (at leas for now).
<img width="1317" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/13470840/bab5cb0e-2dc6-49b8-974f-af3cfcf7873d">